### PR TITLE
[AIP-191] Enforce ruby_package idiomaticity.

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: docs/rules/0191/ruby-package.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+syntax = "proto3";
+
+package google.example.v1beta1;
+
+option ruby_package = "Google::Example::V1beta1"; // Should be V1Beta1.
```
Comment (dazuma) [inline docs/rules/0191/ruby-package.md]: Actually, it _should_ be `V1beta1`, and _not_ `V1Beta`.

Comment (lukesneeringer) [reply docs/rules/0191/ruby-package.md]: Thanks!

Comment (lukesneeringer) [reply docs/rules/0191/ruby-package.md]: Fixed.

File: locations/file_locations.go [hunk 2]
```diff
@@ -43,3 +43,12 @@ func FilePackage(f *desc.FileDescriptor) *dpb.SourceCodeInfo_Location {
 func FileCsharpNamespace(f *desc.FileDescriptor) *dpb.SourceCodeInfo_Location {
 	return pathLocation(f, 8, 37) // 8 == options, 37 == csharp_namespace
 }
+
+// FileRubyPackage returns the location of the csharp_namespace file option
```
Comment (dazuma) [inline locations/file_locations.go]: `s/csharp_namespace/ruby_package/`

File: locations/file_locations.go [hunk 3]
```diff
@@ -43,3 +43,12 @@ func FilePackage(f *desc.FileDescriptor) *dpb.SourceCodeInfo_Location {
...
+// in a file descriptor.
+//
+// If the location can not be found (for example, because there is no
+// csharp_namespace option), it returns nil.
```
Comment (dazuma) [inline locations/file_locations.go]: `s/csharp_namespace/ruby_package/`

Comment (lukesneeringer) [reply locations/file_locations.go]:
```suggestion
// ruby_package option), it returns nil.
```

File: locations/file_locations.go (hunk 2 above)
Comment (lukesneeringer) [reply locations/file_locations.go]:
```suggestion
// FileRubyPackage returns the location of the ruby_package file option
```

**Comment by lukesneeringer** [comment]
> Just want to double-check that Google::UpperCamelExample::V1 is accepted. (Because Google::UpperCamelExample::V1Beta is correctly not accepted, but that is valid upper-camel, and I don't see code that distinguishes that case.) Can you add a test case real quick?

It is accepted (we checked a variant of this in #369), but I am happy to add a test.

One corner case that is _not_ accepted is `Google::Cloud::AutoML::V1`, which it will try to change to `AutoMl`. I think we just have to live with that (we have the ability to disable the rule on an as-needed basis).

**Comment by dazuma** [comment]
Oh. Okay, well, `AutoML` is an uncommon case in Ruby so I guess we can disable the rule in that case.

**Comment by lukesneeringer** [comment]
> Oh. Okay, well, `AutoML` is an uncommon case in Ruby so I guess we can disable the rule in that case.

I added a note about it to the documentation, too.

File: rules/aip0191/ruby_package.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+					Descriptor: f,
+					Location:   locations.FileRubyPackage(f),
+				}}
+			}
+		}
+		return nil
+	},
+}
+
+var rubyValidChars = regexp.MustCompile("^[A-Za-z0-9:]+$")
```
Comment (mingzhi) [inline rules/aip0191/ruby_package.go]: Consider to rename it to `rubyPackageValidCharsRegex`.

Comment (lukesneeringer) [reply rules/aip0191/ruby_package.go]: _Stet._

File: docs/rules/0191/ruby-package.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+package google.example.v1beta1;
+
+option ruby_package = "Google::Example::V1beta1";
+```
+
+## Known issues
+
+This rule will improperly complain if it encounters an acronym. For example, it
+will complain about `Google::Cloud::AutoML::V1`, preferring `AutoMl`. This lint
```
Comment (bufdev) [inline docs/rules/0191/ruby-package.md]: Same comment as on #375
